Update counter test

hakehuang opened the pull request:
using dts to provide board platform level gpt freq workin gpoint instead of hardcode, which is not supported in rt101x and rt102x

MaureenHelm commented:
Can you use the clock control API instead, similar to how LPI2C, LPSPI, and LPUART get the peripheral clocks?

hakehuang commented:
> Can you use the clock control API instead, similar to how LPI2C, LPSPI, and LPUART get the peripheral clocks?

@MaureenHelm  do you mean use below code
```
	clock_dev = device_get_binding(config->clock_name);
	if (clock_dev == NULL) {
		return -EINVAL;
	}

	if (clock_control_get_rate(clock_dev, config->clock_subsys,
				   &clock_freq)) {
		return -EINVAL;
	}
```
instead of using
```
clock_freq = CLOCK_GetFreq(config->clock_source);
```

MaureenHelm commented:
> do you mean use below code

Yes

hakehuang committed:
clocks: ccm add gpt clock control into ccm driver

add clock frequence support for gpt

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

hakehuang committed:
driver: update gpt driver to support internal divider by dts

using dts support gpt clock source and divider

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

hakehuang commented:
> > do you mean use below code
> 
> Yes

@MaureenHelm I update the driver to use the clock from clock system, as there is an internal divider inside gpt, so the nxp,gptfreq attribute is still there, please help to review. Thanks.

MaureenHelm commented on drivers/clock_control/clock_control_mcux_ccm.c:
```diff
@@ -128,6 +128,13 @@ static int mcux_ccm_get_subsys_rate(const struct device *dev,
 	} break;
 #endif
+
+#if CONFIG_COUNTER_MCUX_GPT
```
Please replace `#if` with `#ifdef` for consistency with the rest of the file

hakehuang replied:
ok. done

MaureenHelm commented on dts/bindings/timer/nxp,imx-gpt.yaml:
```diff
@@ -16,3 +16,8 @@ properties:
     label:
       required: true
+
+    nxp,gptfreq:
+      type: int
+      required: true
+      description: gpt frequences
```
@galak it's not clear to me when properties should have a prefix like this

galak replied:
We haven't normally used such prefix in zephyr.  There's some convention of doing this in linux.

MaureenHelm commented on boards/arm/mimxrt1010_evk/mimxrt1010_evk.dts:
```diff
@@ -67,3 +67,11 @@ arduino_serial: &lpuart1 {};
 	status = "okay";
 };
+
+&gpt1 {
+       nxp,gptfreq = <2500000>;
+};
+
+&gpt2 {
+       nxp,gptfreq = <2500000>;
+};
```
Should these be set at the SoC level instead of the board level?

hakehuang replied:
ok, move to SoC level.

MaureenHelm reviewed (changes requested):
@galak please review the new dts property

hakehuang committed:
clocks: add clock api for gpt

add clock api for gpt

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

hakehuang committed:
dts: add nxp gpt freq setting in dts

gpt has internal divider, add dedicated attributes
nxp,gptfreq is added as required one
gpt can customize the gpt freq

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

hakehuang committed:
dts-binding: dts/bindings/timer: add gpt freq in dts binding

add gpt freq in dts binding

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

MaureenHelm reviewed (approved):
Looks good, but I'd like input from @galak about the dts property name.

galak commented:
@MaureenHelm I leave it up to you if you want to merge as is or have @hakehuang tweak the property name to drop the `nxp` prefix.

MaureenHelm reviewed (changes requested):
> @MaureenHelm I leave it up to you if you want to merge as is or have @hakehuang tweak the property name to drop the nxp prefix.

I'd prefer to drop the prefix then

hakehuang commented:
> > @MaureenHelm I leave it up to you if you want to merge as is or have @hakehuang tweak the property name to drop the nxp prefix.
> 
> I'd prefer to drop the prefix then

ok
